platforms/allwinner-d1/src/lib.rs

```diff
@@ -344,22 +337,16 @@ impl D1 {
             }
         }
 
-        timer0.start_counter(0xFFFF_FFFF);
```

Comment:
I wonder if this line moving is why the D1 is wedged. The new code has moved the call to `start_counter` into `Timer0::into_maitake_clock`, which gets called *before* PLIC setup. Perhaps starting the counter before the PLIC is configured breaks something...

Reply:
UPDATE: lol it wasn't that, it was something way stupider: https://github.com/tosc-rs/mnemos/pull/343#issuecomment-2563944528